core/src/main/java/org/apache/iceberg/io/EqualityDeleteWriteResult.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.io;
+
+import java.util.Collections;
+import java.util.List;
+import org.apache.iceberg.DeleteFile;
+
+/**
+ * A result of writing equality delete files.
+ * <p>
+ * Note that objects of this class are NOT meant to be serialized. Task or delta writers will wrap
+ * these results into their own serializable results that can be sent back to query engines.
+ */
+public class EqualityDeleteWriteResult {
```

Comment:
Do we need a special class for this that doesn't have referenced data files? Seems like we could just use the same result for equality and position deletes.

Reply:
I had that in the original PR but then I had to add validation in `RollingEqualityDeleteWriter` and other classes that the result does not have any data files referenced.

I am 50/50 here. I can revert back.

Reply:
I think you are right. I am a bit overthinking it. I'll revert.

Reply:
Reverted.